### PR TITLE
Remove mixed-listed-unlisted waffle and addon unlist function

### PR DESCRIPTION
**Pull request by eviljeff**
last part of #3977, short of actually stopping setting the `is_listed` property on the model.  Could include that too - it would only be another 25 lines or so.

### CODE REVIEW

**Comment by eviljeff** [comment]
Need to wait for the waffle to be flipped on production first to merge this.

**Comment by eviljeff** [comment]
Added 2nd commit that finishes off removing the last few references to `is_listed`, so fixes #3473 too.

File: src/olympia/addons/models.py [hunk 1]
```diff
@@ -352,9 +352,6 @@ class Addon(OnChangeMixin, ModelBase):
 
     whiteboard = models.TextField(blank=True)
 
-    # Whether the add-on is listed on AMO or not.
-    is_listed = models.BooleanField(default=True, db_index=True)
```
Comment (diox) [inline src/olympia/addons/models.py]: We should file an issue to get rid of the column asynchronously. I've checked and dev/stage (presumably prod too, but that needs to be verified before merging) do have a in-database default for this field, so it's fine to remove from the models and do the migration later.

Comment (eviljeff) [inline src/olympia/addons/models.py]: we can leave #3473 open and do it in that.

**Pull request merged**